diff --git a/src/ogg.c b/src/ogg.c
index f8cb0a47..29e7d194 100644
--- a/src/ogg.c
+++ b/src/ogg.c
@@ -734,9 +734,10 @@ ogg_open (SF_PRIVATE *psf)
 			psf->container_data = NULL ;
 			psf->container_close = NULL ;
 			return flac_open (psf) ;
-
+#if 0
 		case SF_FORMAT_OGG | SF_FORMAT_OPUS :
 			return ogg_opus_open (psf) ;
+#endif
 
 #if ENABLE_EXPERIMENTAL_CODE
 		case SF_FORMAT_OGG | SF_FORMAT_SPEEX :
@@ -796,10 +797,11 @@ ogg_stream_classify (SF_PRIVATE *psf, OGG_PRIVATE* odata)
 		case OGG_SPEEX :
 			psf->sf.format = SF_FORMAT_OGG | SF_FORMAT_SPEEX ;
 			return 0 ;
-
+#if 0
 		case OGG_OPUS :
 			psf->sf.format = SF_FORMAT_OGG | SF_FORMAT_OPUS ;
 			return 0 ;
+#endif
 
 		case OGG_PCM :
 			psf_log_printf (psf, "Detected Ogg/PCM data. This is not supported yet.\n") ;
@@ -827,7 +829,9 @@ static struct
 	{	"PCM     ",		"PCM",		8, OGG_PCM },
 	{	"Speex",		"Speex",	5, OGG_SPEEX },
 	{	"\001vorbis",	"Vorbis",	7, OGG_VORBIS },
+#if 0
 	{	"OpusHead",		"Opus",		8, OGG_OPUS },
+#endif
 } ;
 
 static int
diff --git a/src/ogg.h b/src/ogg.h
index 8b2927a9..52947f51 100644
--- a/src/ogg.h
+++ b/src/ogg.h
@@ -28,7 +28,9 @@ enum
 	OGG_PCM,
 	OGG_SPEEX,
 	OGG_VORBIS,
+#if 0
 	OGG_OPUS,
+#endif
 } ;
 
 typedef struct
