--- Csound5.10.1/SConstruct.orig	2008-12-18 18:26:46.000000000 +0000
+++ Csound5.10.1/SConstruct	2009-03-30 20:08:10.000000000 +0100
@@ -768,9 +768,7 @@
 # library version is CS_VERSION.CS_APIVERSION
 csoundLibraryVersion = '5.2'
 csoundLibraryName = 'csound'
-if commonEnvironment['useDouble'] != '0':
-    csoundLibraryName += '64'
-elif getPlatform() == 'win32':
+if getPlatform() == 'win32':
     csoundLibraryName += '32'
 # flags for linking with the Csound library
 libCsoundLinkFlags = []
@@ -814,8 +812,6 @@
             tmp = '%s/lib/csound/plugins' % commonEnvironment['prefix']
         else:
             tmp = '%s/lib64/csound/plugins' % commonEnvironment['prefix']
-        if commonEnvironment['useDouble'] != '0':
-            tmp += '64'
         s = '-DCS_DEFAULT_PLUGINDIR=\\"%s\\"' % tmp
         csoundLibraryEnvironment.Append(CPPFLAGS = [s])
     elif buildOSXFramework != 0:
@@ -2588,10 +2584,7 @@
     if i[:sys.prefix.__len__()] == sys.prefix and i[-13:] == 'site-packages':
         PYTHON_DIR = i
 
-if commonEnvironment['useDouble'] == '0':
-    PLUGIN_DIR = LIB_DIR + "/csound/plugins"
-else:
-    PLUGIN_DIR = LIB_DIR + "/csound/plugins64"
+PLUGIN_DIR = LIB_DIR + "/csound/plugins"
 
 if commonEnvironment['install'] == '1':
     installExecutables = Alias('install-executables',
--- Csound5.10.1/Top/csmodule.c.orig	2008-12-18 17:37:02.000000000 +0000
+++ Csound5.10.1/Top/csmodule.c	2009-03-30 20:08:10.000000000 +0100
@@ -638,17 +638,12 @@
       return CSOUND_ERROR;
 
     /* open plugin directory */
-    dname = csoundGetEnv(csound, (sizeof(MYFLT) == sizeof(float) ?
-                                  plugindir_envvar : plugindir64_envvar));
+    dname = csoundGetEnv(csound, plugindir_envvar);
     if (dname == NULL) {
 #if ENABLE_OPCODEDIR_WARNINGS
       csound->opcodedirWasOK = 0;
-#  ifdef USE_DOUBLE
-      dname = csoundGetEnv(csound, plugindir_envvar);
-      if (dname == NULL)
-#  endif
 #endif
-        dname = CS_DEFAULT_PLUGINDIR;
+      dname = CS_DEFAULT_PLUGINDIR;
     }
     dir = opendir(dname);
     if (UNLIKELY(dir == (DIR*) NULL)) {
--- Csound5.13.0/Top/csmodule.c.orig	2011-04-07 10:48:47.631735127 +0100
+++ Csound5.13.0/Top/csmodule.c	2011-04-07 10:49:19.170735598 +0100
@@ -127,7 +127,6 @@
 
 /* environment variable storing path to plugin libraries */
 static  const   char    *plugindir_envvar =   "OPCODEDIR";
-static  const   char    *plugindir64_envvar = "OPCODEDIR64";
 
 /* default directory to load plugins from if environment variable is not set */
 #if !(defined(_CSOUND_RELEASE_) && (defined(LINUX) || defined(__MACH__)))
